framework/test/unit/modules/interoperability/sidechain/module.spec.ts

```diff
@@ -0,0 +1,546 @@
+
+			await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+				'ownChainNonce must be > 0.',
+			);
+		});
+
+		it('should throw error if chainInfos.length !== 1', async () => {
+			const context = createInitGenesisStateContext(
+				{
+					...defaultData,
+					chainInfos: [chainInfo, chainInfo],
+				},
+				params,
+			);
+			await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+				'chainInfos must contain exactly one entry.',
+			);
+		});
+
+		it('should throw error if mainchainInfo.chainID is not equal to getMainchainID()', async () => {
+			const context = createInitGenesisStateContext(
+				{
+					...defaultData,
+					chainInfos: [
+						{
+							...chainInfo,
+							chainID,
+						},
+					],
+				},
+				params,
+			);
+			await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+				`mainchainInfo.chainID must be equal to ${getMainchainID(chainID).toString('hex')}.`,
+			);
+		});
+
+		describe('chainInfo.chainData', () => {
+			it(`should throw error if chainData.name !== ${CHAIN_NAME_MAINCHAIN}`, async () => {
+				const context = createInitGenesisStateContext(
+					{
+						...defaultData,
+						chainInfos: [
+							{
+								...defaultData.chainInfos[0],
+								chainData: {
+									...chainData,
+								},
+							},
+						],
+					},
+					params,
+				);
+
+				await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+					`chainData.name must be equal to ${CHAIN_NAME_MAINCHAIN}.`,
+				);
+			});
+
+			it('should throw error if chainData.status is not CHAIN_STATUS_REGISTERED or CHAIN_STATUS_ACTIVE', async () => {
+				const context = createInitGenesisStateContext(
+					{
+						...defaultData,
+						chainInfos: [
+							{
+								...defaultData.chainInfos[0],
+								chainData: {
+									...defaultData.chainInfos[0].chainData,
+									status: ChainStatus.TERMINATED,
+								},
+							},
+						],
+					},
+					params,
+				);
+
+				const validStatuses = [ChainStatus.REGISTERED, ChainStatus.ACTIVE];
+				await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+					`chainData.status must be one of ${validStatuses.join(', ')}.`,
+				);
+			});
+
+			it('should throw error if chainData.lastCertificate.timestamp > g.header.timestamp', async () => {
+				const context = createInitGenesisStateContext(
+					{
+						...defaultData,
+						chainInfos: [
+							{
+								...defaultData.chainInfos[0],
+								chainData: {
+									...defaultData.chainInfos[0].chainData,
+									lastCertificate: {
+										...lastCertificate,
+										timestamp: 2000,
+									},
+								},
+							},
+						],
+					},
+					{
+						...params,
+						header: {
+							timestamp: 1000,
+						} as any,
+					},
+				);
+				await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+					'chainData.lastCertificate.timestamp must be < header.timestamp.',
+				);
+			});
+		});
+
+		describe('chainInfo.terminatedStateAccounts', () => {
```

Comment:
Missing tests for `chainInfos.terminatedStateAccounts`: For all entries of the array, `entry.chainID` must be distinct and sorted.

Reply:
```suggestion
		describe('_verifyTerminatedStateAccounts', () => {
```

Reply:
Ok, can be resolved from my side.